openmc/capi/filter.py

```diff
@@ -171,6 +206,20 @@ class EnergyFunctionFilter(Filter):
     filter_type = 'energyfunction'
 
 
+class LegendreFilter(Filter):
```

Comment:
This and the other filter classes need an `__init__` method so that they can be created with an order argument, e.g.
```
f = openmc.capi.LegendreFilter(4)
```

It should look something like:
```Python
def __init__(self, order=None, uid=None, new=True, index=None):
    super().__init__(uid, new, index)
    if order is not None:
        self.order = order
```